packages/server/modules/core/graph/resolvers/users.js

```diff
@@ -1,154 +1,163 @@
 'use strict'
-const appRoot = require( 'app-root-path' )
-const { UserInputError } = require( 'apollo-server-express' )
-const { getUser, getUserByEmail, getUsers, countUsers, getUserRole, updateUser, deleteUser, searchUsers, getUserById, makeUserAdmin, unmakeUserAdmin, archiveUser } = require( '../../services/users' )
-const { saveActivity } = require( `${appRoot}/modules/activitystream/services` )
-const { validateServerRole, validateScopes } = require( `${appRoot}/modules/shared` )
-const zxcvbn = require( 'zxcvbn' )
+const appRoot = require('app-root-path')
+const { UserInputError } = require('apollo-server-express')
+const {
+  getUser,
+  getUserByEmail,
+  getUsers,
+  countUsers,
+  getUserRole,
+  updateUser,
+  deleteUser,
+  searchUsers,
+  getUserById,
+  makeUserAdmin,
+  unmakeUserAdmin,
+  archiveUser
+} = require('../../services/users')
+const { saveActivity } = require(`${appRoot}/modules/activitystream/services`)
+const { validateServerRole, validateScopes } = require(`@/modules/shared`)
+const zxcvbn = require('zxcvbn')
 
 module.exports = {
   Query: {
-
     async _() {
       return `Ph'nglui mglw'nafh Cthulhu R'lyeh wgah'nagl fhtagn.`
     },
 
-    async user( parent, args, context, info ) {
-      await validateServerRole( context, 'server:user' )
+    async user(parent, args, context, info) {
+      // User wants info about himself and he's not authenticated - just return null
+      if (!context.auth && !args.id) return null
```

Comment:
From now on the query will not fail if user is unauthenticated. `null` is a valid value representing that the user isn't logged in